src/System.Memory/src/System/Buffers/ReadOnlySequence_helpers.cs

```diff
@@ -122,7 +116,8 @@ private ReadOnlyMemory<T> GetFirstBuffer()
                 if (typeof(T) == typeof(char) && endIndex < 0)  // SequenceType.String
                 {
                     // No need to remove the FlagBitMask since (endIndex - startIndex) == (endIndex & ReadOnlySequence.IndexBitMask) - (startIndex & ReadOnlySequence.IndexBitMask)
-                    return (ReadOnlyMemory<T>)(object)((string)startObject).AsMemory((startIndex & ReadOnlySequence.IndexBitMask), endIndex - startIndex);
+                    ReadOnlyMemory<char> charMemory = ((string)startObject).AsMemory(startIndex & ReadOnlySequence.IndexBitMask, endIndex - startIndex);
+                    return Unsafe.As<ReadOnlyMemory<char>, ReadOnlyMemory<T>>(ref charMemory);
```

Comment:
Nice. How beneficial is it?

Reply:
Nope, let's not churn for no benefit. Didn't become more readable either.

Reply:
The JIT already special cases this pattern, so I wouldn't expect a performance improvement from it.